### PR TITLE
Select message data in the imap event

### PR DESCRIPTION
**Pull request by luca-angemi**
## Proposed change

Allow to select what data is to be included in the imap event data.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [X] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/115108
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [ ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/imap.markdown [hunk 1]
```diff
@@ -68,6 +68,11 @@ Yahoo also requires the character set `US-ASCII`.
 
 </div>
 
+
+### Selecting message data to include in the imap event (advanced mode)
+
+By default the IMAP event won't include neither `text` nor `headers` message data. If you want either to be included (`text` or `headers` or both) you have to manually select them in the option flow.
+
```
Comment (jbouwh) [inline source/_integrations/imap.markdown]: May be we can also point to use the `imap.fetch` service is the `text` is to be post processed.

**Review by jbouwh — COMMENTED** [review]
https://github.com/home-assistant/home-assistant.io/pull/32212/files#diff-f32ab31fd5ade58f94dbf07510601558acab0d243e9d635f059fc9a23ec0f0b1R128-R129

We should add a note that `text` only will be included if it is explicitly selected in the option flow.

Same applies for headers

**Review by jbouwh — DISMISSED** [review]
LGTM,
Thnx @luca-angemi 👍

**Comment by jbouwh** [comment]
@c0ffeeca7 may be you can have final look?

File: source/_integrations/imap.markdown [hunk 2]
```diff
@@ -68,6 +68,12 @@ Yahoo also requires the character set `US-ASCII`.
 
 </div>
 
+
+### Selecting message data to include in the imap event (advanced mode)
```
Comment (c0ffeeca7) [inline source/_integrations/imap.markdown]:
```suggestion
### Selecting message data to include in the IMAP event (advanced mode)
```

File: source/_integrations/imap.markdown [hunk 3]
```diff
@@ -68,6 +68,12 @@ Yahoo also requires the character set `US-ASCII`.
...
+
+By default the IMAP event won't include neither `text` nor `headers` message data. If you want either to be included (`text` or `headers` or both) you have to manually select them in the option flow.
```
Comment (c0ffeeca7) [inline source/_integrations/imap.markdown]:
```suggestion
By default, the IMAP event won't include `text` or `headers` message data. If you want them to be included (`text` or `headers`, or both), you have to manually select them in the option flow. 
```

File: source/_integrations/imap.markdown [hunk 4]
```diff
@@ -68,6 +68,12 @@ Yahoo also requires the character set `US-ASCII`.
 
 </div>
 
+
+### Selecting message data to include in the imap event (advanced mode)
+
+By default the IMAP event won't include neither `text` nor `headers` message data. If you want either to be included (`text` or `headers` or both) you have to manually select them in the option flow. 
+Another way to process the `text` data is to use the `imap.fetch` service, in this case `text` won't be limited by size.
```
Comment (c0ffeeca7) [inline source/_integrations/imap.markdown]:
```suggestion
Another way to process the `text` data, is to use the `imap.fetch` service. In this case, `text` won't be limited by size.
```

**Review by c0ffeeca7 — APPROVED** [review]

**Pull request merged**